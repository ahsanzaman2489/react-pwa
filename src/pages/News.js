import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import * as NewsActions from '../actions/newsActions';
import * as qs from 'query-string';

// import {NO_DATA} from '../../constants/app';


export class NewsPage extends Component {
    state = {
        query: this.props.location.search
    };

    componentDidMount() {
        const {location, fetchNewsSources} = this.props;
        const query = qs.parse(location.search);
        fetchNewsSources('sources', '&category=' + query.category);
    }

    componentWillReceiveProps(nextProps) {
        const {location, fetchNewsSources} = nextProps;
        const query = qs.parse(location.search);


        if (this.state.query !== location.search) {
            fetchNewsSources('sources', '&category=' + query.category);
            this.setState({query: location.search})
        }
    }


    render() {
        return (

            <div className="car-detail">
                News
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    console.log(state)
    return {
        sources: state.headlineReducers,
    };
};

function mapDispatchToProps(dispatch) {
    return {...bindActionCreators(NewsActions, dispatch)}
}


export default connect(mapStateToProps, mapDispatchToProps)(NewsPage);
