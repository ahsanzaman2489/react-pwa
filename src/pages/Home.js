import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import * as NewsActions from '../actions/News';

// import {NO_DATA} from '../../constants/app';


class HomePage extends Component {
    state = {

    };

    componentDidMount() {

    }


    render() {

        return (

            <div className="car-detail">
                Home
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        car: state,
    };
};

function mapDispatchToProps(dispatch) {
    return {...bindActionCreators({NewsActions}, dispatch)}
}


export default connect(mapStateToProps, mapDispatchToProps)(HomePage);
