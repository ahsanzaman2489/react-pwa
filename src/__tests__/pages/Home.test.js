import React from 'react';
import {shallow} from 'enzyme/build';
import {HomePage} from '../../pages/Home';


const setup = (props = {}, state = null) => {
    const defaultProps = {
        fetchHeadLines: jest.fn(),
        headlines: {articles: []}
    };
    const setUpProps = {...defaultProps, ...props};
    const wrapper = shallow(<HomePage {...setUpProps}/>);
    const tree = () => setTimeout(() => renderer.create(<HomePage {...setUpProps}/>).toJSON());
    if (state) wrapper.setState(state);
    return {wrapper, props: setUpProps, tree}
};

let wrapper, props;

beforeEach(() => {
    const setupWrapper = setup();
    wrapper = setupWrapper.wrapper;
    props = setupWrapper.props;
});
describe("Home component", () => {
    it('Should render', () => {
        console.log(wrapper.debug())
        expect(wrapper.find('CardColumns').length).toBe(1)
        expect(wrapper.find('h1').length).toBe(1)

        expect(props.fetchHeadLines).toHaveBeenCalled()
        expect(props.fetchHeadLines).toHaveBeenCalledTimes(1);
    });
});

