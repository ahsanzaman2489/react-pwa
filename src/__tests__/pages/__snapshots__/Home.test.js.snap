// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match with snapshot 1`] = `3`;
